### ISSUE TITLE
dotnet restore NullReferenceException

### ISSUE THREAD

**Issue by joshfree** [dotnet/cli#750]
Filing on behalf of @mafshin from https://github.com/dotnet/coreclr/issues/2499

Following [Getting Started](https://dotnet.github.io/getting-started) Console app with .NET Core just after `dotnet new` I get a `NullReferenceException` for `dotnet restore`:

```
D:\Test>dotnet restore
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Dnx.Tooling.Program.Main(String[] args)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Dnx.Runtime.Common.EntryPointExecutor.Execute(Assembly assembly, String[] args, IServiceProvider serviceProvider)
   at Microsoft.Dnx.Host.Bootstrapper.RunAsync(List`1 args, IRuntimeEnvironment env, String appBase, FrameworkName targetFramework)
   at Microsoft.Dnx.Host.RuntimeBootstrapper.ExecuteAsync(String[] args, BootstrapperContext bootstrapperContext)
   at Microsoft.Dnx.Host.RuntimeBootstrapper.Execute(String[] args, BootstrapperContext bootstrapperContext)
```

**Comment by livarcocc** [comment]
Which OS is this?!

**Comment by mafshin** [comment]
Windows 10 64 bit

**Comment by piotrpMSFT** [comment]
whoa, this is interesting. Can you check %DOTNET_HOME%\bin\dotnet-restore[.exe|.cmd] and tell me which extension you see?

**Comment by mafshin** [comment]
It's a CMD file

On Fri, Jan 8, 2016 at 9:49 AM, Piotr Puszkiewicz <notifications@github.com>
wrote:

> whoa, this is interesting. Can you check
> %DOTNET_HOME%\bin\dotnet-restore[.exe|.cmd] and tell me which extension you
> see?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dotnet/cli/issues/750#issuecomment-169908546>.
>



-- 
"Mathematics is the queen of the sciences and number theory is the queen of
mathematics."
--Gauss

**Comment by piotrpMSFT** [comment]
ok. We recently converted it from CMD to EXE and I thought there might be a regression... how old is your installation? What happens if you call DNU restore directly? 

My expectation [hope] is that we have an older install with an older DNU. I think we had NullRefs in DNU early on in the project...

**Comment by mafshin** [comment]
It's not very old, maybe a week ago.

Calling `dnu restore` directly starts listing dependencies.

Should I download and install the latest [dotnet-win](https://dotnetcli.blob.core.windows.net/dotnet/dev/Installers/Latest/dotnet-win-x64.latest.msi) ?

**Comment by piotrpMSFT** [comment]
yea, I think that's the right move. I'm not able to repro locally...

**Comment by mafshin** [comment]
Using the old one, after `dun restore` I was able to run the app via `dotnet run` but `dotnet restore` was not working.

I deleted the old project completely and installed the new dotnet-win package. I can now see the `dotnet-restore.exe` file in the home directory.

![dotnet](https://cloud.githubusercontent.com/assets/4930000/12193419/6911370e-b5fd-11e5-8df0-37191d2c48fc.PNG)

Yet again after installing the latest MSI package, I observe the same behavior, `dotnet restore` throws a `NullReferenceException` but with slightly more information this time. It seems `Microsoft.DotNet.ProjectModel.Project.get_Tools()` can't be found. 

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Dnx.Tooling.Program.Main(String[] args)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Dnx.Runtime.Common.EntryPointExecutor.Execute(Assembly assembly, String[] args, IServiceProvider serviceProvider)

   at Microsoft.Dnx.Host.Bootstrapper.RunAsync(List`1 args, IRuntimeEnvironment env, String appBase, FrameworkName targetFramework)
Method not found: 'System.Collections.Generic.List`1<Microsoft.DotNet.ProjectModel.Graph.LibraryRange> Microsoft.DotNet.ProjectModel.Project.get_Tools()'.
   at Microsoft.Dnx.Host.RuntimeBootstrapper.ExecuteAsync(String[] args, BootstrapperContext bootstrapperContext)
   at Microsoft.Dnx.Host.RuntimeBootstrapper.Execute(String[] args, BootstrapperContext bootstrapperContext)
```

**Comment by piotrpMSFT** [comment]
This should be fixed with #421. If it repros after that gets checked in, please reopen this issue @mafshin
